src/portalbackend/PortalBackend.PortalEntities/Auditing/AuditHandlerV1.cs

```
@@ -0,0 +1,119 @@
+                                 .Select(x => x.Name),
+                             property => property.Metadata.Name))
+                {
+                    prop.CurrentValue = _identityService.IdentityData.UserId;
+                }
+
+                // If existing try to set DateLastChanged
+                foreach (var prop in entry.Properties.Where(x => x.Metadata.Name == "DateLastChanged"))
+                {
+                    prop.CurrentValue = _dateTimeProvider.OffsetNow;
+                }
+            }
+
+            AddAuditEntry(entry, entry.Metadata.ClrType, context);
+        }
+    }
+
+    private void AddAuditEntry(EntityEntry entityEntry, Type entityType, DbContext context)
+    {
+        var auditEntityAttribute = (AuditEntityV1Attribute?)Attribute.GetCustomAttribute(entityType, typeof(AuditEntityV1Attribute));
+        if (auditEntityAttribute == null)
+        {
+            throw new ConfigurationException($"{entityType} must be annotated with {nameof(AuditEntityV1Attribute)}");
+        }
+
+        var auditEntityType = auditEntityAttribute.AuditEntityType;
+        var sourceProperties = new List<PropertyInfo>();
+        if (typeof(IBaseEntity).IsAssignableFrom(entityType))
+        {
+            sourceProperties.AddRange(typeof(IBaseEntity).GetProperties());
+        }
+        sourceProperties.AddRange(entityType.GetProperties(BindingFlags.Public |
+                                                           BindingFlags.Instance |
+                                                           BindingFlags.DeclaredOnly).Where(p => !(p.GetGetMethod()?.IsVirtual ?? false)));
+        var auditProperties = typeof(IAuditEntityV1).GetProperties();
+        var targetProperties = auditEntityType.GetProperties().ExceptBy(auditProperties.Select(x => x.Name), p => p.Name);
+
+        if (Activator.CreateInstance(auditEntityType) is not IAuditEntityV1 newAuditEntity)
+            return;
```

this should never happen as AuditEntityV1Attribute can only be used on types implementing IAuditEntityV1. As an assertion this has not been changed throw UnexpectedConditionException instead of returning silently.